### PR TITLE
Uplift third_party/tt-metal to 10d5c0c4276958e4bb582829d0d33cd222205d7a 2025-08-02

### PR DESCRIPTION
**Pull request by vmilosevic**
This PR uplifts the third_party/tt-metal to the 10d5c0c4276958e4bb582829d0d33cd222205d7a

### CODE REVIEW

**Review by mtopalovicTT — APPROVED** [review]

**Review by nsmithtt — APPROVED** [review]

File: tools/ttnn-standalone/CMakeLists.txt [hunk 1]
```diff
@@ -115,6 +115,8 @@ set(INCLUDE_DIRS
     ${METAL_SRC_DIR}/ttnn/core
     ${METAL_SRC_DIR}/ttnn/cpp
     ${METAL_SRC_DIR}/ttnn/cpp/ttnn
+
+    $ENV{TT_METAL_HOME}/build/include
```
Comment (svuckovicTT) [inline tools/ttnn-standalone/CMakeLists.txt]: What was missing that this was needed? Would prefer to include from source instead of build.

Comment (mtopalovicTT) [reply tools/ttnn-standalone/CMakeLists.txt]: So they decided to use enchantum instead of magic_enum and they did it by using FetchContent instead of CPM (not sure what is the reasoning).
I'm trying to unblock the build, but it seems that its still failing so it might be good to see if we can make them switch to CPM.

https://github.com/tenstorrent/tt-metal/pull/25367

Comment (svuckovicTT) [reply tools/ttnn-standalone/CMakeLists.txt]: Looks like the install script is failing, because of this: https://github.com/tenstorrent/tt-mlir/blob/cffc895f22ceedd1fee6244b8d32ac1c2dfcfb1a/third_party/CMakeLists.txt#L183

Comment (svuckovicTT) [reply tools/ttnn-standalone/CMakeLists.txt]: Though I agree, using cpm would make it so much easier. Otherwise we'll need more involved changes to keep FFE working with emitc.

Comment (mtopalovicTT) [reply tools/ttnn-standalone/CMakeLists.txt]: Thats not the issue unfortunately. @afuller-TT can we use cmp for enchantum also? Fetch content is not putting stuff in build/include until actual build is performed, and we have targets which link against metal headers which use enchantum and do not build metal fully but rely on metal-configure step to pull the dependencies using .cpm.
Long story short, it's complicated and it would be lot easier if you can use cpm :)

Comment (mtopalovicTT) [reply tools/ttnn-standalone/CMakeLists.txt]: https://github.com/tenstorrent/tt-metal/pull/26257

I've created a change in metal to change it to CPM since it would be hard to change how we build different targets to how things currently are.

File: runtime/lib/ttnn/program_executor.cpp [hunk 2]
```diff
@@ -361,7 +361,7 @@ void ProgramExecutor::dumpPerfCountersIfNeeded() {
     LOG_DEBUG(LogType::LogRuntimeTTNN, "Dumping device profile results after " +
                                            std::to_string(counter) +
                                            " operations");
-    ::tt::tt_metal::DumpMeshDeviceProfileResults(context->getMeshDevice());
+    ::tt::tt_metal::ReadMeshDeviceProfilerResults(context->getMeshDevice());
```
Comment (brataTT) [inline runtime/lib/ttnn/program_executor.cpp:364]: Cherry-picked from @mtopalovicTT 's branch `milant/manualMetalUpdate`.

**Comment by brataTT** [comment]
<details><summary>Brings 70 metal commits</summary>

```j
> git log d78b2f9ae3d33c468d71c99ee5e8ab7f3606c89d..8e131829e4f91371a8dd59c0c4a6fa83ce13a87e --format="%cd %h by %an|||: %s" --date=short-local | column -t -s "|||"
2025-08-01 8e131829e4 by Pavlo Hilei                : Fix integer overflow in std::accumulate (#26070)
2025-08-01 40e485deac by Debin Chen                 : #16352 Adding broadcasted tensors gives wrong results (#26101)
2025-08-01 7d751099a4 by Billy Ransom               : Support Asymmetric Padding in Generic Pools (#25994)
2025-08-01 944d34c651 by Venkatesh Guduru           : #25002: restructure trace and add DP support(n300) for mobilenet_v2 (#25067)
2025-08-01 474059b906 by Aswin Zayasankaran         : Migrate CELU as LLK operation (#25872)
2025-08-01 48799b947d by Colman Glagovich           : Enable MUX in SD3.5 CCLs (#25947)
2025-08-01 8cf10daec7 by Keerthana                  : #25175: Add DP Support for Yolov7 (#25408)
2025-08-01 3a5395eec5 by Keerthana                  : #24618: Add DP Support for Yolov8x (#24625)
2025-08-01 b63abae491 by Stuti Raizada              : Accept temp, top_k, top_p ttnn tensor inputs in ttnn.sampling (#25540)
2025-08-01 a3b9e8d918 by Pratikkumar Prajapati      : LMHead implementation of deepseek (#26043)
2025-08-01 6de880f56c by Ammar Vora                 : MLA Meta-Layer Support (#26029)
2025-08-01 4894a8bbc4 by Keerthana                  : #24912: Add DP Support for Yolov8s_World (#24913)
2025-08-01 3313a94b2d by Djordje Ivanovic           : Llama70B topK op - unit test update (#26122)
2025-08-01 6d151e7ae5 by Oleg Milyutin              : #25469: Multi-host aware device reshapes (#25833)
2025-08-01 fdb52ea035 by Mouliraj Elamurugan        : #21829: Migrate softsign as device op (#25942)
2025-08-01 4a72832c13 by Nikola Stamatovic          : [DM] Add loopback directed ideal test (#26059)
2025-08-01 7d5b570096 by Aleksandar Stancov         : Conv2d skip_mcast cleanup (#25889)
2025-08-01 83691992d9 by Pavlo Hilei                : Optimize nd resharding op + Reenable DRAM nd resharding (#25902)
2025-08-01 592f7a0849 by Slavko Krstic              : #26001: Support tile dest offset in pack untilize dest (#26002)
2025-08-01 223b16cd38 by Stuti Raizada              : Llama TG: seqlen 128 ttft improvements  (#26017)
2025-08-01 3c913deea6 by Aswin Zayasankaran         : chore: update LLK submodule to 29cc2c7 (#26107)
2025-08-01 41ee87c2ad by David Popov                : Remove qwen from nightly and qwen25_vl (#26114)
2025-08-01 8d7b03bcb6 by Gongyu Wang                : Qwen2.5-VL bringup (#24236)
2025-08-01 82d9e85f81 by Stanislav Minakov          : #25975: Migrate pad, permute, repeat, reshape OPs to use TensorAccessor (#26091)
2025-08-01 5f25d3c7e5 by Vincent Tang               : TensorAccessorArgs pybinds (#25979)
2025-07-31 750a1ee20a by Grayson Rechsteiner        : RS Intermediate Buffer Cleanup (#25801)
2025-07-31 76b41ae2e8 by Atul Krishnadas            : enable int32 on reshape (#25585)
2025-07-31 c772b5f8db by Ammar Vora                 : Add Paged FlashMLA Prefill support (#26080)
2025-07-31 efc40d478b by Samarth Agarwal            : Rename profiler-related functions, variables, comments and documentation from 'Dump' to 'Read' across the entire codebase (#25954)
2025-07-31 1cb9bf3684 by William Ly                 : Revert "Replace truncating fp32->bf16 with tie-to-even rounding rule" (#26098)
2025-07-31 4645848ad6 by Samarth Agarwal            : Avoid calling MetalContext in DeviceProfiler destructor (#26074)
2025-07-31 10a29abea4 by Keerthana Rajendran        : #18636: Yolov12x ttnn bring-up (#24569)
2025-07-31 6db05009f9 by William Ly                 : #25958: [skip ci] Clean malformed XML if parsing fails (#25956)
2025-07-31 078962d266 by Marvin Mok                 : EnqueueWaitForEvent Test Migration to use TT-Mesh APIs, Test Fixture Cleanup (#25710)
2025-07-31 522be6231c by Saad Jameel                : Add Fabric 1D ring support for all_to_all_dispatch (#25847)
2025-07-31 ebcfbec674 by Johanna Rock               : Rename llama3_subdevices to llama3_galaxy (#25885)
2025-07-31 b1305e23e0 by Kartik Paigwar             : api change fix for deepseek mla ops (#26094)
2025-07-31 a654b649cf by Yousef Al Rawwash          : Resolving Llama 70B model device perf hang on TG and 6U (#25978)
2025-07-31 98292b3708 by Nour Ardo                  : Fix CI failure for SD and falcon (#26064)
2025-07-31 564a4bbe3f by Rose Li                    : [skip ci] Add option to specify commit to custom test dispatch pipeline (#26072)
2025-07-31 3456a55ccd by Pavlo Hilei                : Replace truncating fp32->bf16 with tie-to-even rounding rule (#19882)
2025-07-31 e75f2d4108 by Stanislav Minakov          : #26044: Fix ct args passed to element wise kernel after TensorAccessor refactoring (#26047)
2025-07-31 d5a40c57b3 by William Ly                 : [skip ci] Move N300 ttnn unit tests back to CIv1 (#26089)
2025-07-31 53e9784464 by Bartłomiej Cieślar         : MLP and RMSNorm metamodules (#26009)
2025-07-31 db31954d39 by Iva Potkonjak              : SDXL: VAE untilize removed (#26063)
2025-07-31 bc29843833 by Colman Glagovich           : SD3.5 - parallelize CLIP + T5 and integrate into pipeline (#26016)
2025-07-31 a6655f7bab by Sudharsan Vijayaraghavan   : Vovnet model bringup (#21441)
2025-07-31 dea866a742 by Mouliraj Elamurugan        : #24693: Migrate hardswish as a device op (#25857)
2025-07-31 59436a1500 by David Popov                : Skip tutorial tests on BH because of test hanging (#26069)
2025-07-31 7b93ae96a0 by Aleksandar Stancov         : lowered expected yolo9c perf (#26054)
2025-07-31 7ad1f359f7 by Pavle Josipović            : SDXL Conv2d tweak float32 acc settings (#26018)
2025-07-31 90b58bdab9 by Pavlo Hilei                : Kernel code indexing (#25739)
2025-07-31 4ae09fcfbb by Sabira Shaik               : ttnn bringup for efficicnetnetb0 model (#21429)
2025-07-31 9623460011 by Stanislav Minakov          : #25975: Port all tt_dnn kernels to use TensorAccessor (#26041)
2025-07-31 c02cf58765 by Miguel Tairum              : Update N150 Llama-3.2-1B targets (#26021)
2025-07-31 e61a9b767e by Aleksandar Stancov         : bumped bert_tiny perf, lowered mnist perf (#26046)
2025-07-31 93212a6166 by Aleksandar Djordjevic      : Enabling check_noc_status to be run from triage (#25944)
2025-07-31 08294a65d6 by Sudharsan Vijayaraghavan   :  #21550: Yolov5x ttnn bringup (#25007)
2025-07-31 686ea5d25a by Marko Staletovic           : Added TILE input tensor support for interleaved nearest upsample (#25901)
2025-07-31 35272b56cc by Miłosz Gajewski            : #25614: Add ttnn tutorials tests under minimum venv (#25874)
2025-07-31 7f587cd8fa by Aswin Zayasankaran         : Migrate xlogy to LLK (#25001)
2025-07-30 221891bae5 by Kannika Kabilar            : Fix Single-card Device perf report (#26034)
2025-07-30 f2881a1417 by Bryan Wilder Field Lozano  : [skip ci] Don't run test-wheels job in package and release (#26042)
2025-07-30 e8da6d830e by Stanislav Minakov          : Fix Blackhole APC tests related to recent migration to TensorAccessor (#26037)
2025-07-30 86c16e9094 by Aniruddha Tupe             : Fixed failing tests for SBert on T3K (#26032)
2025-07-30 b46b164bf0 by Debin Chen                 : Revert "#23626 Binary_ng row subtile broadcast using LLK with bfloat1… (#26035)
2025-07-30 2344ba5775 by Nour Ardo                  : fix untilize blackhole failure (#26026)
2025-07-30 a7485d312c by Stanislav Minakov          : #25975: Migrate gather, indexed_fill, move, non_zero_indicies OPs to use TensorAccessor (#25995)
2025-07-30 16308ac20a by Stanislav Minakov          : #25975: Migrate clone, copy, fill, fold OPs to use TensorAccessor (#25987)
2025-07-30 7daa65926e by Pratikkumar Prajapati      : Add test for real weights (#25926)
```

</details>


FE CI runs:
- [x] [tt-torch CI](https://github.com/tenstorrent/tt-torch/actions/runs/16782646892) 
- [x] [tt-forge-fe CI](https://github.com/tenstorrent/tt-forge-fe/actions/runs/16782956433) (fails are xpasses and huggingface rate limits)
- [x] [tt-xla CI](https://github.com/tenstorrent/tt-xla/actions/runs/16782963122)

**Comment by kmabeeTT** [comment]
Nice job, @brataTT, I'll force merge this in a few mins

File: test/ttmlir/Silicon/TTNN/n150/pooling/avg_pool2d.mlir [hunk 3]
```diff
@@ -46,16 +46,16 @@ module {
   }
 
   // Test 4: AvgPool2dOp with flattened compat info and with kernel, stride, padding and ceil_mode specified
```
Comment (brataTT) [inline test/ttmlir/Silicon/TTNN/n150/pooling/avg_pool2d.mlir:48]: Thanks for fast unblock for pooling here @sdjordjevicTT . The failing test helped identify [[an edge case]](https://github.com/tenstorrent/tt-metal/commits/fba3e6cca73658a0874cf44d70982fdecadfc423) in tt-metal too (though we don't hit it from tt-mlir tests)

File: runtime/include/tt/runtime/detail/ttmetal/ttmetal.h [hunk 4]
```diff
@@ -101,7 +101,7 @@ void deallocateBuffers(Device device);
 
 void dumpMemoryReport(Device device);
 
-void dumpDeviceProfileResults(Device device);
+void readDeviceProfilerResults(Device device);
```
Comment (mtopalovicTT) [inline runtime/include/tt/runtime/detail/ttmetal/ttmetal.h:104]: @tapspatel Not sure if you want to align python bindings for this runtime calls, I forgot to rewrite them.
dump_device_profile_results -> read_device_profiler_results

File: test/ttmlir/Silicon/TTNN/n150/pooling/avg_pool2d.mlir (hunk 3 above)
Comment (sdjordjevicTT) [reply test/ttmlir/Silicon/TTNN/n150/pooling/avg_pool2d.mlir:48]: Great! I think we will need to incorporate this as well on the next uplift once we consume their fix.

File: runtime/include/tt/runtime/detail/ttmetal/ttmetal.h (hunk 4 above)
Comment (brataTT) [reply runtime/include/tt/runtime/detail/ttmetal/ttmetal.h:104]: python bindings for this one should be covered [[down here]](https://github.com/tenstorrent/tt-mlir/pull/4324/files#diff-e17ad9668e939b91d767c7226c9a0e6b836a06230d89452a2768462acc895e40R44)
Frontends don't refer to it, so no change was needed downstream

Comment (mtopalovicTT) [reply runtime/include/tt/runtime/detail/ttmetal/ttmetal.h:104]: I missed that PR somehow. Tnx!
